src/com/rcv/Tabulator.java

```diff
@@ -158,11 +159,15 @@ public void tabulate() {
       // Conversely, the currentRoundCandidateToTally object returned here will contain fewer
       // entries, each of which will have as many or more votes than they did in prior rounds.
       // Eventually the winner(s) will be chosen.
-      Map<String, Float> currentRoundCandidateToTally = getTallyForRound(currentRound);
+      Map<String, BigDecimal> currentRoundCandidateToTally = getTallyForRound(currentRound);
       roundTallies.put(currentRound, currentRoundCandidateToTally);
 
+      // cache this as it will change after adding winners to winnerToRound
+      // TODO: better encapsulation
```

Comment:
getThreshold depends on previous winners which is why it needs to be "cached" here (before previous winners is modified).  Since determineWinners calculates threshold as part of it's process I think it might be cleaner if determineWinners also returns that threshold it used for that round.